cluster/get-kube-binaries.sh

```diff
@@ -0,0 +1,199 @@
+      echo "Bailing out." >&2
+      exit 3
+      ;;
+  esac
+}
+
+function md5sum_file() {
+  if which md5 >/dev/null 2>&1; then
+    md5 -q "$1"
+  else
+    md5sum "$1" | awk '{ print $1 }'
+  fi
+}
+
+function sha1sum_file() {
+  if which shasum >/dev/null 2>&1; then
+    shasum -a1 "$1" | awk '{ print $1 }'
+  else
+    sha1sum "$1" | awk '{ print $1 }'
+  fi
+}
+
+function download_tarball() {
+  local -r download_path="$1"
+  local -r file="$2"
+  url="${DOWNLOAD_URL_PREFIX}/${file}"
+  mkdir -p "${download_path}"
+  if [[ $(which curl) ]]; then
+    curl -L "${url}" -o "${download_path}/${file}"
+  elif [[ $(which wget) ]]; then
+    wget "${url}" -O "${download_path}/${file}"
+  else
+    echo "Couldn't find curl or wget.  Bailing out." >&2
+    exit 4
+  fi
+  echo
+  local md5sum=$(md5sum_file "${download_path}/${file}")
+  echo "md5sum(${file})=${md5sum}"
+  local sha1sum=$(sha1sum_file "${download_path}/${file}")
+  echo "sha1sum(${file})=${sha1sum}"
+  echo
+  # TODO: add actual verification
+}
+
+function extract_tarball() {
+  local -r tarfile="$1"
+  local -r platform="$2"
+  local -r arch="$3"
+
+  platforms_dir="${KUBE_ROOT}/platforms/${platform}/${arch}"
+  echo "Extracting ${tarfile} into ${platforms_dir}"
+  mkdir -p "${platforms_dir}"
+  # Tarball looks like kubernetes/{client,server}/bin/BINARY"
+  tar -xzf "${tarfile}" --strip-components 3 -C "${platforms_dir}"
+  # Create convenience symlink
+  ln -sf "${platforms_dir}" "$(dirname ${tarfile})/bin"
+  echo "Add '$(dirname ${tarfile})/bin' to your PATH to use newly-installed binaries."
+}
+
+detect_kube_release
+
+SERVER_PLATFORM="linux"
+SERVER_ARCH="${KUBERNETES_SERVER_ARCH:-amd64}"
+SERVER_TAR="kubernetes-server-${SERVER_PLATFORM}-${SERVER_ARCH}.tar.gz"
+
+detect_client_info
+CLIENT_TAR="kubernetes-client-${CLIENT_PLATFORM}-${CLIENT_ARCH}.tar.gz"
+
+echo "Kubernetes release: ${KUBERNETES_RELEASE}"
+echo "Server: ${SERVER_PLATFORM}/${SERVER_ARCH}"
+echo "Client: ${CLIENT_PLATFORM}/${CLIENT_ARCH}"
+echo
+
+# TODO: remove this check and default to true when we stop shipping server
+# tarballs in kubernetes.tar.gz
+DOWNLOAD_SERVER_TAR=false
+if [[ ! -e "${KUBE_ROOT}/server/${SERVER_TAR}" ]]; then
+  DOWNLOAD_SERVER_TAR=true
+  echo "Will download ${SERVER_TAR} from ${DOWNLOAD_URL_PREFIX}"
+fi
+
+# TODO: remove this check and default to true when we stop shipping kubectl
+# in kubernetes.tar.gz
+DOWNLOAD_CLIENT_TAR=false
+if [[ ! -x "${KUBE_ROOT}/platforms/${CLIENT_PLATFORM}/${CLIENT_ARCH}/kubectl" ]]; then
+  DOWNLOAD_CLIENT_TAR=true
+  echo "Will download and extract ${CLIENT_TAR} from ${DOWNLOAD_URL_PREFIX}"
+fi
+
+if [[ "${DOWNLOAD_CLIENT_TAR}" == false && "${DOWNLOAD_SERVER_TAR}" == false ]]; then
+  echo "Nothing additional to download."
+  exit 0
+fi
+
+if [[ -z "${KUBERNETES_SKIP_CONFIRM-}" ]]; then
+  echo "Is this ok? [Y]/n"
+  read confirm
+  if [[ "${confirm}" =~ ^[nN]$ ]]; then
+    echo "Aborting."
+    exit 0
+  fi
+fi
+
+if "${DOWNLOAD_SERVER_TAR}"; then
```

Comment:
Perhaps echo the tar file URLs you would download?  This will help in the "offline" case where we can tell folks what they should download somehow.